### ISSUE TITLE
Use GoF Bridge pattern to decouple compiler implementation from its use

### ISSUE THREAD

**Issue by ghost** [CrispOSS/jabsc#19]
This would 
1) allow for future flexibility
2) make compiler use easier for tools ( e.g. a maven plugin, or an IDE )
3) completely hide all compilation details to a user
4) allow us to work on improved versions / algos without causing breakage to users ( humans or tools )

![bridgedcompiler](https://cloud.githubusercontent.com/assets/2629689/8206675/f7a4dc94-14fb-11e5-83ca-74bb6351d046.jpg)

**Comment by nobeh** [comment]
I agree with the general design patterns. As per Maven, we already have a Maven plugin supporting jabsc at: https://github.com/CrispOSS/jabsc-maven-plugin and example usage at https://github.com/CrispOSS/jabsc-maven-plugin-example
